test/users.spec.js

```diff
@@ -71,93 +76,174 @@ describe('User', () => {
       done();
     });
   });
 
   it('should validate non-admin users cannot get all users.', (done) => {
-    request.get('/api/users').set('x-access-token', token).set('Accept', 'application/json')
-    .expect(401).end((err, res) => {
+    request.get('/api/users').set('x-access-token', token)
+    .set('Accept', 'application/json')
+    .expect(401)
+    .end((err, res) => {
       expect(res.body.message).to.equal('Access denied.');
       done();
     });
   });
 
+  it('should report when user does not exist.', (done) => {
+    request.get('/api/users/barma').set('x-access-token', token)
+    .set('Accept', 'application/json')
+    .expect(401)
+    .end((err, res) => {
+      expect(res.body.message).to.equal('Valid user name required.');
+      done();
+    });
+  });
+
   it('should validate that users can update details.', (done) => {
     request.put('/api/users/adlaw').set('x-access-token', token).set('Accept', 'application/json').send({
       lastname: 'Lawal',
       email: 'adLawal@testing.com'
-    }).expect(200).end((err, res) => {
+    })
+    .expect(200)
+    .end((err, res) => {
       expect(res.body.lastname).to.equal('Lawal');
       expect(res.body.email).to.equal('adLawal@testing.com');
       done();
     });
   });
 
   it('should ensure non-admin users can only update their profiles.', (done) => {
-    request.put('/api/users/admin').set('x-access-token', token).set('Accept', 'application/json').send({
+    request.put('/api/users/admin').set('x-access-token', token)
+    .set('Accept', 'application/json').send({
       username: 'admina'
-    }).expect(401).expect({message: 'Access denied.'}).end(done);
+    })
+    .expect(401)
+    .expect({ message: 'Access denied.' })
+    .end(done);
+  });
+
+  it('should ensure only previously created users can update their profiles.', (done) => {
+    request.put('/api/users/adminat').set('x-access-token', adminToken)
+    .set('Accept', 'application/json').send({
+      username: 'adminana'
+    })
+    .expect(400)
+    .expect({ message: 'Username or email is invalid.' })
+    .end(done);
+  });
+
+  it('should ensure correct username and password are required for login.', (done) => {
+    request.post('/api/users/login').set('Accept', 'application/json').send({
+      username: 'admin',
+      password: 'Panting'
+    })
+    .expect(401)
+    .expect({ message: 'Wrong Password.' })
+    .end(done);
+  });
+
+  it('should show when user does not exist for login.', (done) => {
+    request.post('/api/users/login').set('Accept', 'application/json').send({
+      username: 'adminat',
+      password: 'Panting'
+    })
+    .expect(404)
+    .expect({ message: 'User does not exist.' })
+    .end(done);
   });
 
   it('should ensure non-admin users can only update their profiles.', (done) => {
-    request.put('/api/users/tutu').set('x-access-token', adminToken).set('Accept', 'application/json').send({
+    request.put('/api/users/tutu').set('x-access-token', adminToken)
```

Comment:
### Critical

This test case says `non-admin` users meanwhile the JWT is an `admin token`, how come?